### PR TITLE
Initializer Part II, adding the initializer

### PR DESCRIPTION
**Pull request by caesarxuchao**
Adding the initializer object, which discovers migratable resources, and creates StorageVersionMigraion objects for each migratable resources.

### CODE REVIEW

File: pkg/initializer/initializer.go [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+}
+
+func migrationForResource(resource schema.GroupVersionResource) *migrationv1alpha1.StorageVersionMigration {
+	var name string
+	if len(resource.Group) != 0 {
+		name = fmt.Sprintf("%s.%s.%s-", resource.Group, resource.Version, resource.Resource)
+	} else {
+		name = fmt.Sprintf("%s.%s-", resource.Version, resource.Resource)
+	}
+	return &migrationv1alpha1.StorageVersionMigration{
```
Comment (yliaog) [inline pkg/initializer/initializer.go]: do you want to add a schema for it?

File: pkg/initializer/initializer.go [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+	})
+	if err != nil {
+		return err
+	}
+	_, err = init.crdClient.Create(migrationCRD())
+	return err
+}
+
+// TODO: remove this function. Users will use provided yaml files to create the
+// namespace, and then create the initializer in the namespace.
```
Comment (yliaog) [inline pkg/initializer/initializer.go]: it seems that the whole CRD creation etc could be done out of band, like what you suggested above for namespace

File: pkg/initializer/initializer.go (hunk 1 above)
Comment (caesarxuchao) [reply pkg/initializer/initializer.go]: I'll add it in a follow-up.

**Pull request merged**